source/aggregation-tutorials/group-total.txt

```diff
@@ -0,0 +1,225 @@
+            stage, calculate the following new fields:
+
+            - ``first_purchase_date``: the date of the customer's first purchase
+            - ``total_value``: the total value of all the customer's purchases
+            - ``total_orders``: the total number of the customer's purchases
+            - ``orders``: the list of all the customer's purchases,
+              including the date and value of each purchase
+
+            .. literalinclude:: /includes/aggregation/group-total.js
+               :language: javascript
+               :copyable: true
+               :start-after: start-group
+               :end-before: end-group
+               :dedent:
+
+         .. step::
+
+            Next, add another :manual:`$sort
+            </reference/operator/aggregation/sort>` stage to set an
+            ascending sort on the ``first_purchase_date`` field:
+
+            .. literalinclude:: /includes/aggregation/group-total.js
+               :language: javascript
+               :copyable: true
+               :start-after: start-sort2
+               :end-before: end-sort2
+               :dedent:
+
+         .. step::
+
+            Add a :manual:`$set
+            </reference/operator/aggregation/set>` stage to recreate the
+            ``customer_id`` field from the values in the ``_id`` field
+            that were set during the ``$group`` stage:
+
+            .. literalinclude:: /includes/aggregation/group-total.js
+               :language: javascript
+               :copyable: true
+               :start-after: start-set
+               :end-before: end-set
+               :dedent:
+
+         .. step::
+
+            Finally, add an :manual:`$unset
+            </reference/operator/aggregation/unset>` stage. The
+            ``$unset`` stage removes the ``_id`` field from the result
+            documents:
+            
+            .. literalinclude:: /includes/aggregation/group-total.js
+               :language: javascript
+               :copyable: true
+               :start-after: start-unset
+               :end-before: end-unset
+               :dedent:
+
+   .. step:: Run the Aggregation Pipeline
+
+      Run the following command in your shell to perform the aggregation
+      on the ``products`` collection:
+
+      .. code-block:: bash
+      
+         node agg_tutorial.js
+
+      The aggregated result contains three documents. The documents
+      describe customers and a summary of their orders:
+
+      .. code-block:: javascript
+         :copyable: false
+         
+         {
+           first_purchase_date: 2020-01-01T08:25:37.000Z,
+           total_value: 63.13,
+           total_orders: 1,
+           orders: [ { orderdate: 2020-01-01T08:25:37.000Z, value: 63.13 } ],
+           customer_id: 'oranieri@warmmail.com'
+         }
+         {
+           first_purchase_date: 2020-01-13T09:32:07.000Z,
+           total_value: 482.15999999999997,
```

Comment:
Issue:

I think including rounding errors can cause confusion and even distrust of the product.

Suggestion:
I saw a comment in another tutorial PR that mentioned that this data is from the book, but I think this could be a case in which we would want to use data that doesn't generate floating point approximation errors like this one.

Reply:
Sounds good, I will revise to use whole numbers only